source/elements/oneMKL/source/domains/spblas/data_types/create_coo_matrix.rst

```diff
@@ -0,0 +1,256 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_create_coo_matrix:
+
+create_coo_matrix
+=================
+
+Initializes a ``matrix_handle_t`` object with the provided COO data.
+
+.. rubric:: Description and Assumptions
+
+The ``oneapi::mkl::sparse::create_coo_matrix`` function initializes the
+``matrix_handle_t`` object with the provided data.
+
+In the case of buffers, the reference count of the provided buffer is
+incremented which extends the lifetime of the underlying buffer until the sparse
+matrix handle is destroyed with ``destroy_sparse_matrix`` or the data is reset
+with ``set_coo_matrix_data``.
+
+In the case of USM, the object does not take ownership of the data.
+
+The ``oneapi::mkl::sparse::create_coo_matrix`` function defined below takes in
+the number of non-zero elements in the sparse matrix as an argument. However, in
+certain math operations where the output is a sparse matrix, e.g., sparse matrix
+addition (sparse matrix + sparse matrix = sparse matrix), and multiplication of
+two sparse matrices, the number of non-zero elements in the output sparse matrix
+is not known in advance and must be calculated as part of the operation API.
+Such APIs are currently not a part of the current oneMKL Specification, but will
+be added in the future. Therefore, it is important for the ``create_coo_matrix``
+API to be able to handle multiple calls to build the sparse matrix as function
+arguments become known and arrays are allocated with the correct sizes. In
+particular, the ``create_coo_matrix`` API must handle the case where the number
+of non-zero elements in the matrix are unknown and the matrix arrays are
+``nullptr`` or zero-sized ``sycl::buffer`` objects as applicable. This behavior
+is currently left to be implementation-defined, but may be clarified in the
+oneMKL Specification in the future.
+
+See :ref:`onemkl_sparse_matrix_handle`.
+
+.. _onemkl_sparse_create_coo_matrix_buffer:
+
+create_coo_matrix (Buffer version)
+----------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void create_coo_matrix (sycl::queue                          &queue,
```

Comment:
why not return matrix_handle_t from the function instead of passing it in if it should be uninitialized?

naming consideration: `create_coo_matrix` or `make_coo_matrix`? (e.g. make_tuple in C++)

Reply:
There is an internal host-side allocation made by and owned by the library in this API. There may be a preference for `create`/`init` over `make` because of that. cuSPARSE/rocSPARSE use `create`, Intel oneMKL has `init` and `set`.

Reply:
I considered returning the handle but decided against it after all. There were still some doubts on whether:

1. These functions should take and return SYCL events
2. Allow to take in an already initialized handle.

So far we decided against it but keeping the existing API is safer if this changes in the future.